lib/statsd/instrument/assertions.rb

```diff
@@ -44,21 +44,33 @@ module StatsD::Instrument::Assertions
 
   # Asserts no metric occurred during the execution of the provided block.
   #
-  # @param [String] metric_name (default: nil) The metric name that is not allowed
-  #   to happen inside the block. If this is set to `nil`, the assertion will fail
-  #   if any metric occurs.
+  # @param [Array<String, Regexp>] metric_names_or_pattern (default: []) The metric
+  #   names, or name patterns that are not allowed to happen inside the block. If
+  #   this is set to `[]`, the assertion will fail if any metric occurs.
   # @yield A block in which the specified metric should not occur. This block
   #   should not raise any exceptions.
   # @return [void]
   # @raise [Minitest::Assertion] If an exception occurs, or if any metric (with the
-  #   provided name, or any), occurred during the execution of the provided block.
-  def assert_no_statsd_calls(metric_name = nil, datagrams: nil, client: nil, &block)
+  #   provided names, patterns, or any), occurred during the execution of the
+  #   provided block.
+  def assert_no_statsd_calls(*metric_names_or_patterns, datagrams: nil, client: nil, &block)
     if datagrams.nil?
       raise LocalJumpError, "assert_no_statsd_calls requires a block" unless block_given?
       datagrams = capture_statsd_datagrams_with_exception_handling(client: client, &block)
     end
 
-    datagrams.select! { |m| m.name == metric_name } if metric_name
+    unless metric_names_or_patterns.empty?
+      datagrams.select! do |metric|
+        metric_names_or_patterns.any? do |metric_name_or_pattern|
+          case metric_name_or_pattern
+          when Regexp
+            metric_name_or_pattern =~ metric.name
+          when String
+            metric_name_or_pattern == metric.name
+          end
```

Comment:
I originally implemented this as
```ruby
metric_name_or_pattern === metric.name
```
but Rubocop complained.

Technically, this implementation means that if someone was previously using something other than a `String`, it would no longer work, but I'm not sure how that would have been working in the first place, or if it should be supported. I could always switch to something like

```ruby
case metric_name_or_pattern
when Regexp
  metric_name_or_pattern =~ metric.name
else
  metric_name_or_pattern == metric.name
end
```
or
```ruby
if metric_name_or_pattern.is_a?(Regexp)
  metric_name_or_pattern =~ metric.name
else
  metric_name_or_pattern == metric.name
end
```